### PR TITLE
(variables) - regex change for variables prefixed with dollars

### LINKED ISSUE #1782: Dollar prefix with a variable in text bubble does not appear
https://discord.com/channels/1155799591220953138/1281842592056868894/1281842592056868894

### PR DESCRIPTION
**Pull request by ahsmha**
Fixes #1782

### CODE REVIEW

**Comment by baptisteArno** [comment]
Not good you ignore the case `${{{Variable}}}`

**Comment by ahsmha** [comment]
@baptisteArno what could be the appropriate fix over here?
and when is ${{{var}}} case used?

**Comment by ahsmha** [comment]
One workaround is that we could store dollar sign inside a variable and then use it alongside the value without space in between. But that is just a workaround

File: packages/variables/parseVariables.ts [hunk 1]
```diff
@@ -23,6 +23,9 @@ const inlineCodeRegex = /\{\{=(.+?)=\}\}/g
 // {{variable}} and ${{{variable}}}
 const variableRegex = /\{\{([^{}]+)\}\}|(\$)\{\{([^{}]+)\}\}/g
```
Comment (baptisteArno) [inline packages/variables/parseVariables.ts:24]: You basically ignored that regex. We need to extend it instead of replacing it with a simpler one

**Comment by CptPlastic** [comment]
I look at the `$` as currency so maybe just use a bunch of them and that is the variable 

you could set a list. and only accept that in your regex. I fear that in some instances the $ will break things and would be best practice to wrap currency in its own variable 
```
Kč
₡
RD$
$
```

File: packages/variables/parseVariables.ts (hunk 1 above)
Comment (ahsmha) [reply packages/variables/parseVariables.ts:24]: ```\{\{([^{}]+)\}\}|(\$)\{\{\{([^{}]+)\}\}\}```
@baptisteArno what do you think of this one?
so for input - 
```
${{xyz}} 
${{{abc}}} 
{{pqr}}
```

it gives following matches - 
```
{{xyz}}
${{{abc}}}
{{pqr}}
```

i wonder if it breaks anything?

Comment (baptisteArno) [reply packages/variables/parseVariables.ts:24]: I fixed the issue already 🙏

https://github.com/baptisteArno/typebot.io/commit/3c07041f2f0a03852138b953a51838b81b077895

**Pull request closed without merging**